CONTRIBUTING.rst

```diff
@@ -0,0 +1,86 @@
+Contributing
+=========================
+
+As an open source project, Mesa welcomes contributions of many forms, and from beginners to experts. 
+
+Contributions can a full model or it could one of the following to an existing model: 
+
+- Code patches
+- Bug reports and patch reviews
+- New features
+- Documentation improvements
+- Tutorials
+
+No contribution is too small. Although, contributions can be too big, so let's
+discuss via `Matrix`_ OR via `an issue`_.
+
+.. _`Mesa discussions`: https://github.com/projectmesa/mesa/discussions
+.. _`Matrix`: https://matrix.to/#/#project-mesa:matrix.org`
+.. _`an issue` : https://github.com/projectmesa/mesa-examples/issues
+
+**To submit a contribution**
+
+- Create a ticket for the item that you are working on.
+- Fork the mesa-examples repository.
+- `Clone your repository`_ from Github to your machine.
+- Create a new branch in your fork: ``git checkout -b BRANCH_NAME``
+- Run ``git config pull.rebase true``. This prevents messy merge commits when updating your branch on top of Mesa main branch.
+- If you are using the existing Mesa version that a model might be pinned to, then you will want to run ``pip install -r requirements.txt`` from inside the example folder you are working with. If you are making changes using the main repo or other version, do not install the version in the requirements file & make sure to update if the example is pinned.
```

Comment:
This line specifically could use some finessing. People with less Python experience are more likely to contribute a model than core code. But here is a situation where they need to pay attention to the versions of things. That is what I was trying to describe. Maybe we pass this thru for now, but reapproach when we do a rewrite of this as I suggest here: https://github.com/projectmesa/mesa/issues/963#issuecomment-1336722683